### PR TITLE
Stop default recorder if failed to start a message

### PR DESCRIPTION
**Pull request by ihorserba**
The default recorder kept recording once failed to start a stream

### CODE REVIEW

File: sdks/js/src/classes/outgoingMessage.js [hunk 1]
```diff
@@ -173,6 +173,11 @@ outgoingMessage.then(function(result) {
       this.currentMessageId = result.stream_id;
       this.startRecording();
     }).catch((err) => {
+      // The default recorder started once created
```
[the remark this answers is not in the document]
Comment (ihorserba) [reply sdks/js/src/classes/outgoingMessage.js:176]: [It's misleading](https://github.com/zelloptt/zello-channel-api/blob/2e8d8c21ef9bf6447c08002b87baaf8743cd9e26/sdks/js/src/classes/recorder.js#L228) for the default recorder

[the remark this answers is not in the document]
Comment (ihorserba) [reply sdks/js/src/classes/outgoingMessage.js:176]: But it still works for a custom one with the proper `start()` method

[the remark this answers is not in the document]
Comment (vocoded) [reply sdks/js/src/classes/outgoingMessage.js:176]: i see, that's unfortunate. the comment doesn't seem correct though, the default recorder starts when `init()` is called, not on creation

[the remark this answers is not in the document]
Comment (ihorserba) [reply sdks/js/src/classes/outgoingMessage.js]: Yeah, I meant the OutgoingMessage creation.
Changed the comment

File: sdks/js/src/classes/outgoingMessage.js [hunk 2]
```diff
@@ -173,6 +173,12 @@ outgoingMessage.then(function(result) {
       this.currentMessageId = result.stream_id;
       this.startRecording();
     }).catch((err) => {
+      /* The default recorder is started on OutgoingMessage instance creation
+       * once recorder's init() method is called */
+      this.stopRecording();
+
+      /* TODO: There is no easy way to catch this async exception.
+       * Implement better error handling */
```
Comment (vocoded) [inline sdks/js/src/classes/outgoingMessage.js:181]: this method should return the promise

**Review by vocoded — APPROVED** [review]

File: sdks/js/src/classes/outgoingMessage.js (hunk 2 above)
Comment (ihorserba) [reply sdks/js/src/classes/outgoingMessage.js:181]: Yes, and probably we have to pass a callback to the constructor to handle the error

Comment (vocoded) [reply sdks/js/src/classes/outgoingMessage.js:181]: @ihorserba i was suggesting to make the change now - it shouldn't break any existing clients

Comment (ihorserba) [reply sdks/js/src/classes/outgoingMessage.js:181]: Oh, I just saw it has been approved, so I merged it.
I'll add a new PR with these changes

**Pull request merged**